frontend/src/components/common/ChangePasswordModal.tsx

```diff
@@ -0,0 +1,248 @@
+                  setSubmitted(true);
+                  setError(true);
+                }
+              });
+          })
+          .catch((e) => {
+            if ((e as FirebaseError).code === "auth/wrong-password") {
+              setFeedback("The password you entered is incorrect.");
+            } else {
+              setFeedback(
+                "An error occurred, please refresh the page and try again.",
+              );
+            }
+          });
+      } else {
+        setFeedback(
+          "An error occurred, please refresh the page and try again.",
+        );
+      }
+    }
+  };
+
+  if (submitted && !error) {
+    return (
+      <Modal
+        onClose={() => {
+          onClose();
+          reset();
+        }}
+        isOpen={isOpen}
+        size="lg"
+        isCentered
+      >
+        <ModalOverlay />
+        <ModalContent padding="10px">
+          <ModalCloseButton />
+          <ModalBody>
+            <Heading size="lg" mt="40px" mb="40px">
+              Password changed!
+            </Heading>
+            <Text mb="220px">
+              You have successfully changed your password. You may now close
+              this window and return to your work
+            </Text>
+            <Image
+              src={PasswordChanged}
+              w="70%"
+              position="absolute"
+              bottom="0"
+              right="0"
+              borderRadius="8px"
+            />
+          </ModalBody>
+        </ModalContent>
+      </Modal>
+    );
+  }
+
+  if (submitted && error) {
+    return (
+      <Modal
+        onClose={() => {
+          onClose();
+          reset();
+        }}
+        isOpen={isOpen}
+        size="lg"
+        isCentered
+      >
+        <ModalOverlay />
+        <ModalContent padding="10px">
+          <ModalCloseButton />
+          <ModalBody>
+            <Heading size="lg" mt="40px" mb="40px">
+              An error has occurred
+            </Heading>
+            <Text mb="160px">
+              Uh oh! Something went wrong with changing your password. Please
+              try again later. <br />
+              <br /> If the error persists, try refreshing the page and
+              clearning your browser of cookies.
+            </Text>
+            <Image
+              src={PasswordError}
+              w="70%"
+              position="absolute"
+              bottom="0"
+              right="0"
+              borderRadius="8px"
+            />
+          </ModalBody>
+        </ModalContent>
+      </Modal>
+    );
+  }
+
+  return (
+    <Modal
+      onClose={() => {
+        onClose();
+        reset();
+      }}
+      isOpen={isOpen}
+      size="lg"
+      isCentered
+    >
+      <ModalOverlay />
+      <ModalContent padding="10px">
+        <ModalHeader>Change password</ModalHeader>
+        <ModalCloseButton />
+        <ModalBody>
+          <Text fontWeight="bold" mb="8px">
+            Old password
+          </Text>
+          <Input
```

Comment:
Can you use the PasswordInputField.tsx component here? It is found in the common components folder. Ik the designs didn't include this component, but I think they should to keep the UX consistent based on other password components in the app too? @susheelk @desginers

Reply:
+1 on this. That component also gives us nice things like the "eye" to toggle visibility, turning the field red on invalidity, etc.